### DISCUSSION TITLE
updateListing when passing taxonomy_id triggers error

### DISCUSSION THREAD

**Discussion by jcroucher** [etsy/open-api#1053] [Q&A]
Calling updateListing with taxonomy_id gives an error Bad request: {"error":"There was a problem with \/value : cannot be empty"}

I note there are inconsistencies with the api docs which could be part of my issue

The main code I am referencing is from which looks like the v3 docs
https://developers.etsy.com/documentation/reference/#operation/updateListing

When I do a PATCH request to

https://openapi.etsy.com/v3/application/shops/XX/listings/XX

It returns {"error"=>"Resource not found"}

But the item does exist.
I then found the docs for what I think is the old v2.

https://www.etsy.com/developers/documentation/reference/listing#method_updatelisting

Even though I am using the v3 endpoint, I noticed this one uses a PUT instead of PATCH.
Hitting this endpoint works, but it then returns a
{"error":"There was a problem with \/value : cannot be empty"}

I see that the two docs have slightly different fields, so I updated them to match the correct PUT vs PATCH but it still returns {"error":"There was a problem with \/value : cannot be empty"}

There is then a tutorials page which gives an example. 
https://developer.etsy.com/documentation/tutorials/listings

It uses PUT but has a different endpoint URL

https://api.etsy.com/v3/application/shops
vs
https://openapi.etsy.com/v3/application/shops/

I can do a createDraftListing request with all the fields which as no error, but if I pass the same fields with the new listing ID it triggers the error. I went through one by one to narrow it down to taxonomy_id

**Comment by emergingdzns** [comment]
The "Resource Not Found" error is actually an error with the URL or method. It means that their system can't find the destination you're sending the data to. It has nothing to do with the listing itself. I would say check your URL very carefully to make sure that it is in fact a PATCH (the PUT method is deprecated in favor of the PATCH method and documentation in the tutorials never got updated) and that the path it is sending to is correct. Also verify that you are sending as "application/x-www-form-urlencoded" data.

It might help if you posted your curl payload (minus your auth token and api keys of course) so that we can review what is being sent and to what url.

**Comment by jcroucher** [comment]
The resource not found was json vs urlencoded. Thanks for that
The main issue is still there on this endpoint though

**URL**: https://openapi.etsy.com/v3/application/shops/XX/listings/XX

**Headers**: {"x-api-key":"XX","Authorization":"Bearer XX","Content-Type":"application/x-www-form-urlencoded"}

**Method**: PATCH

**Payload**: {"title":"Example T-Shirt", "taxonomy_id": 374}

**Response**: Error: Bad request: {"error":"There was a problem with \/value : cannot be empty"}

If I remove taxonomy_id from the update request it works.
